src/net_processing.cpp

```diff
@@ -1691,6 +1692,7 @@ bool static ProcessHeadersMessage(CNode *pfrom, CConnman *connman, const std::ve
                 }
                 pindexWalk = pindexWalk->pprev;
             }
+            CHECK(pindexWalk);
```

Comment:
Looks like you are using this wrong. This should be an assert, since it is dereferenced later on.

Reply:
I agree it is not an optimal example: `assert(…);` would obviously be the more appropriate choice for cases where we want an unconditional `abort` if the assumption does not hold. I'll remove this example.